### ISSUE TITLE
The size of emailed screenshots of the same image varies with browser window size

### ISSUE THREAD

**Issue by scottwittenburg** [OpenImaging/miqa#99]
From a user:

> It seems that the sizing of the attached snapshots for the same session vary in relationship to the client browser’s window size.  e.g. I’ve got a huge 4k display… if I maximize the browser and send screenshots, the resulting image attachments are very large.  However if I shrink the browser window down to ¼ of the display, I get much smaller attachments.  I would expect to get the same sized image irrespective of the browser window dimensions. Is there a way we can make this consistent irrespective of the browser size.

**Comment by curtislisle** [comment]
How about adding a SimpleITK processing step in the email handler somewhere around https://github.com/OpenImaging/miqa/blob/master/server/miqa_server/email.py#L46 to resample the image to 512x512 before appending it to the HTML?

**Comment by scottwittenburg** [comment]
That's an option.  But I'm also wondering if we can find api somewhere on the client that would let us generate the image at a specified size.  These values could either be 512 x 512 as you suggested, or else exposed to the user as an option as time permits.

**Comment by curtislisle** [comment]
I agree that the size should be configurable either by the user or by a configuration option somewhere in the future. It is OK by me for the size to be hard-coded at this time for the purposes of this operational test by Stanford.

**Comment by scottwittenburg** [comment]
Looping in @jimklo in case he has thoughts.

**Comment by scottwittenburg** [comment]
So far I haven't found api in vtk.js that would let us request a screenshot at a specified size.

Regardless of the approach we end up taking in the long run, I think I still have some questions about how this should behave.  

1. I assume we want the screenshot to maintain the aspect ratio of the original image?  If so, how should we choose an image size?  E.g. we could choose to fix the larger dimension of screenshots to certain size, or fix the width to a certain size, and let the other dimension vary with the aspect ratio of the original image.
2. What kind of interpolation (for growing) and what kind of sampling (for shrinking) should we use in resizing?
3. Should the screenshot dimensions be chosen based on the original image resolution (perhaps to avoid excessive interpolation/sampling)?

**Comment by curtislisle** [comment]
Great questions.  I think we should keep the longer side of the image equal to 512 ( which is often the native resolution) and resize the shorter side to keep the aspect ratio.  I would use BILINEAR resampling, which is supported by the PIL.Image.resize method.  We should be able to do all with PIL without having to go to ITK. In many cases, we won’t have to resample much, I hope.

**Comment by curtislisle** [comment]
Per discussion with Kilian & Jim on 4/28 - They indicated that aspect ratio has to be maintained.  Their use case is that sometimes images are too large.  Images almost/never need to be made larger.    We discussed that MIQA (in the longer term) could be configurable somewhere to have default image sizes.  In the current version, I believe we can set a fixed, reasonable size (e.g. 512xYYY).  I suggest we implement either in the client or server.  Looking at the server code, it looks not too hard to add an ITK filter in the Email section of the server code.

**Comment by aashish24** [comment]
@scottwittenburg from Sebastien

1) screenshot same size to replace the rw with it

2) hide and resize rw

3) take screenshot

4) resize rw to origin and put it back to its original location

**Comment by aashish24** [comment]
As per our discussions @scottwittenburg will push a PR to vtk.js to address MIQA needs (which also benefits others efforts) and if time remains integrate it in MIQA or hand if off to someone else.

**Comment by jimklo** [comment]
I've reviewed #114 in our [integration environment](https://sibis.sri.com/miqa).  The unless there's some additional setting that needs to be done, this doesn't appear to work quite yet.  Via email all images I receive appear to be significantly larger than 512x512 (which is the fixed size hard coded in the PR.

**Comment by dzenanz** [comment]
#114 is a work in progress, in a sense that it depends on https://github.com/Kitware/vtk-js/pull/1907.

**Comment by aashish24** [comment]
I will follow up on this issue and revert back here
